src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
```

Comment:
> At the very least I feel that your new change should use the common pattern even if we create a new issue to clean up the rest of the code.

I agree with @PaulHigin. Using the recommended pattern in your new code can serve as a good example for anyone who'd come later to fix the rest of code.